src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -5,493 +5,121 @@
-        const handlerFileRelativePath = path.relative(
-            rootCodeFolder,
-            path.dirname(this.localInvokeArgs.document.uri.fsPath)
-        )
-
-        const relativeFunctionHandler = path.join(
-            handlerFileRelativePath,
-            this.localInvokeArgs.handlerName
-        ).replace('\\', '/')
-
-        const workspaceFolder = vscode.workspace.getWorkspaceFolder(this.localInvokeArgs.workspaceFolder.uri)
-        let existingTemplateResource: CloudFormation.Resource | undefined
-        if (workspaceFolder) {
-            const lambdas = await detectLocalLambdas([workspaceFolder])
-            const existingLambda = lambdas.find(lambda => lambda.handler === relativeFunctionHandler)
-            existingTemplateResource = existingLambda ? existingLambda.resource : undefined
-        }
-
-        let newTemplate = new SamTemplateGenerator()
-            .withCodeUri(rootCodeFolder)
-            .withFunctionHandler(relativeFunctionHandler)
-            .withResourceName(LocalLambdaRunner.TEMPLATE_RESOURCE_NAME)
-            .withRuntime(this.runtime)
-
-        if (existingTemplateResource && existingTemplateResource.Properties &&
-            existingTemplateResource.Properties.Environment) {
-            newTemplate = newTemplate.withEnvironment(existingTemplateResource.Properties.Environment)
-        }
-
-        await newTemplate.generate(inputTemplatePath)
-
-        return inputTemplatePath
-    }
-
-    private async determineRootCodeFolder(): Promise<string> {
-        const packageJsonPath: string | undefined =
-            await filesystemUtilities.findFileInParentPaths(
-                path.dirname(this.localInvokeArgs.document.uri.fsPath),
-                'package.json'
-            )
-
-        if (!packageJsonPath) {
-            throw new Error(
-                localize(
-                    'AWS.error.sam.local.package_json_not_found',
-                    'Unable to find package.json related to {0}',
-                    this.localInvokeArgs.document.uri.fsPath
-                )
-            )
-        }
-
-        return path.dirname(packageJsonPath)
-    }
-
-    private async executeSamBuild(
-        rootCodeFolder: string,
-        inputTemplatePath: string
-    ): Promise<string> {
-        this.outputChannel.appendLine(
-            localize(
-                'AWS.output.building.sam.application',
-                'Building SAM Application...'
-            )
-        )
-
-        const samBuildOutputFolder = path.join(await this.getBaseBuildFolder(), 'output')
-
-        await new SamCliBuildInvocation(
-            samBuildOutputFolder,
-            rootCodeFolder,
-            inputTemplatePath,
-            this.processInvoker
-        ).execute()
-
-        this.outputChannel.appendLine(
+import {  TypescriptLambdaHandlerSearch } from '../typescriptLambdaHandlerSearch'
+
+export const getSamNodeProjectDirPath = async (): Promise<string> => {
```

Comment:
`getSamNodeProjectDirPath` should be able to work as follows:
input: a js file's path
output: the project dir

It shouldn't need an active Text Editor, and callers should know what file they are interested in finding the project dir for.

Reply:
It's not used by anything else and does what it needs to for it's current use... but I can see that there might eventually be broader use cases. Renamed to `getSamProjectDirPathForFile` and used same name and interface for both CodeLens providers. If this were to be re-used, it likely belongs elsewhere... Made it private.
✔